lib/internal/process/warning.js

```diff
@@ -127,6 +127,11 @@ function onWarning(warning) {
 // process.emitWarning(str[, type[, code]][, ctor])
 // process.emitWarning(str[, options])
 function emitWarning(warning, type, code, ctor) {
+  // Fast path to avoid memory allocation,
+  // this doesn't eliminate the other if a few lines below
+  if (type === 'DeprecationWarning' && process.noDeprecation) {
```

Comment:
Shouldn't you invert the checks so we can skip the string comparison when `noDeprecation` is false?

Reply:
Make sense